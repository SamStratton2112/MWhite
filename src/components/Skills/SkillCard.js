
import { Col, Container} from 'react-bootstrap';
import {useState} from 'react';

const SkillCard = ({name, image}) => {
    const [hidden, setHidden] = useState(true);
    const body2 = <h4>{name}</h4>;
    const body1 = <img  style={{height:'15em', borderRadius:'.2em'}} src={image} alt="Skateboarding"/>;
    return (
        <Col>
            <Container 
                style={{ width: '15em', height: '15em', border: 'solid', margin: '1em', borderRadius:'.3em', textAlign:'center', backgroundColor:'rgb(184, 17, 17, .5)'}}
                className='center' 
                onMouseEnter={() => setHidden(false)} 
                onMouseLeave={() => setHidden(true)}
            >
                {hidden ? body1 : body2}
            </Container>
        </Col>
    );
};

export default SkillCard;