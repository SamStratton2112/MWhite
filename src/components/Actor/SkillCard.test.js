import SkillCard from './SkillCard';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

describe(SkillCard, () => {
    it("renders without crashing", () => {
        render(<MemoryRouter>
            <SkillCard />
        </MemoryRouter>);
    });
    it("matches snapshot", () => {
        const { asFragment } = render(<MemoryRouter>
            <SkillCard />
        </MemoryRouter>);
        expect(asFragment()).toMatchSnapshot();
    });
});