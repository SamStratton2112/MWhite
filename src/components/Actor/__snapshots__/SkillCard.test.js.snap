// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="col"
  >
    <div
      class="center container"
      style="width: 15em; height: 15em; border: solid; margin: 1em; border-radius: .3em; text-align: center;"
    >
      <img
        alt="Skateboarding"
        style="height: 15em; border-radius: .2em;"
      />
    </div>
  </div>
</DocumentFragment>
`;
